common/net/bolt1.py

```diff
@@ -0,0 +1,283 @@
+            return InitMessage.from_bytes(message)
+        elif mtype == message_types["error"]:
+            return ErrorMessage.from_bytes(message)
+        elif mtype == message_types["ping"]:
+            return PingMessage.from_bytes(message)
+        elif mtype == message_types["pong"]:
+            return PongMessage.from_bytes(message)
+        else:
+            raise ValueError("Cannot decode unknown message type")
+
+    def serialize(self):
+        """Serialises the message."""
+        if not self.extension:
+            return self.type + self.payload
+        else:
+            tlvs = b"".join([tlv.serialize() for tlv in self.extension])
+            return self.type + self.payload + tlvs
+
+
+class InitMessage(Message):
+    """
+    First message exchange by the nodes, it reveals the features supported by each end.
+
+    Args:
+        global_features (:obj:`FeatureVector <teos.net.bolt9.FeatureVector>`): the global features vector.
+        local_features (:obj:`FeatureVector <teos.net.bolt9.FeatureVector>`): the local features vector.
+        local_features (:obj:`list`): a list of genesis block hashes (optional).
+    """
+
+    def __init__(self, global_features, local_features, networks=None):
+        if not (isinstance(global_features, FeatureVector) and isinstance(local_features, FeatureVector)):
+            raise TypeError("global_features and local_features must be FeatureVector instances")
+        if networks:
+            if not isinstance(networks, NetworksTLV):
+                raise TypeError("networks must be of type NetworksTLV (if set)")
+
+        gf = global_features.serialize()
+        lf = local_features.serialize()
+        gflen = len(gf).to_bytes(2, "big")
+        flen = len(lf).to_bytes(2, "big")
+        payload = gflen + gf + flen + lf
+
+        # Add extensions if needed (this follows TLV format)
+        # FIXME: Only networks for now
+        if networks:
+            super().__init__(mtype=message_types["init"], payload=payload, extension=[networks])
+        else:
+            super().__init__(mtype=message_types["init"], payload=payload)
+        self.global_features = global_features
+        self.local_features = local_features
+        self.networks = networks
+
+    @classmethod
+    def from_bytes(cls, message):
+        """Builds an InitMessage from its byte representation."""
+        # Message should be at least: type (2-byte) + gflen (2-byte) + flen (2 byte)
+        message_sanity_checks(message, message_types["init"], 6)
+
+        try:
+            gflen = int.from_bytes(message[2:4], "big")
+            global_features = FeatureVector.from_bytes(message[4 : gflen + 4])
+            flen = int.from_bytes(message[gflen + 4 : gflen + 6], "big")
+            local_features = FeatureVector.from_bytes(message[gflen + 6 : gflen + flen + 6])
+            if gflen + flen + 6 > len(message):
+                raise ValueError()  # Unexpected EOF
```

Comment:
It it rather harmless though.

If `flen` is smaller than what it is supposed to, it may either:

- Create an invalid `FeatureVector`, which will raise a `ValueError`.
- Create a valid `FeatureVector` but with insufficient length, which will raise this trigger check, and therefore raise a `ValueError`.

I get you comment tho, but I went for this approach for readability given the outcome was the same.